misc/package/Data Files/MWSE/core/mcm/components/settings/Setting.lua

```diff
@@ -72,4 +72,71 @@ function Setting:convertToLabelValue(variableValue)
 	return variableValue
 end
 
+-- Check if the keys of `t1` are a subset of the keys of `t2`.
+local function isSubset(t1, t2)
+	-- Something is a subset if it's not not a subset
+    for k in pairs(t1) do
+        if t2[k] == nil then return false end
+    end
+    return true
+end
+
+-- Finds a default value, given a variable. 
+-- This function handles nested subtables, but it is sometimes incorrect.
+---@param var mwseMCMTableVariable
+---@param defaultTable table A table nested inside a `defaultConfig`. 
+-- This could be equal to `defaultConfig` itself
+---@return any defaultValue
+local function findRecursive(var, defaultTable)
+
+    -- A loaded `configTable` could have keys that are not in the default table.
+    -- But, if it was loaded with `mwse.loadConfig`, then every key in `configTable` will be in `defaultTable`.
+	-- So, the subset check is a heuristic that helps to increase the likelihood of picking the correct subtable.
+    if defaultTable[var.id] ~= nil and isSubset(defaultTable, var.table) then
+        return defaultTable[var.id]
+    end
+
+	-- Key not in `defaultTable`, or `defaultTable` wasn't a good match? Check subtables.
+    for _, defaultVal in pairs(defaultTable) do
+        if type(defaultVal) == "table" then
+            local res = findRecursive(var, defaultVal)
+            if res ~= nil then 
+                return res
+            end
+        end
+    end
+end
+
+--- This will recursively go through your MCM and append the text "Default = ___" to the description of each setting.
+---@param defaultConfig table? the default config of your mod. if not provided, it will try to be retrieved, 
+-- using the path "config.default"
+function Setting:addDefaultsToDescriptions(defaultConfig)
+    if not defaultConfig then return end
+
+	local var = self.variable
+	if not var or var.class ~= "TableVariable" then return end
+	---@cast var mwseMCMTableVariable
+
+	local defaultVal = var.defaultSetting
+	if defaultVal == nil then
+		defaultVal = findRecursive(var, defaultConfig)
```

Comment:
After some discussion with Merlord (which you can see on our Discord channel), I think it would be best not to do the heuristic finding of the default values from the `defaultConfig` table. Only use the `defaultSetting` field provided by the variable.